src/DataTable/CardView.jsx

```diff
@@ -1,22 +1,51 @@
-import React from 'react';
+import React, { useContext } from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
+import DataTableContext from './DataTableContext';
 import { useRows } from './hooks';
 import { CardGrid } from '..';
 
+function CardItem({
+  row,
+  prepareRow,
+  isSelectable,
+  SelectionComponent,
+  CardComponent,
+  selectionPlacement,
+}) {
+  prepareRow(row);
```

Comment:
[curious] Should the call to `prepareRow` be in a `useEffect` (or otherwise) to only have it called once (or only when `row` changes)? I'd wonder if there's any concern around it being called too many times?

Reply:
I don't think `useEffect` is very useful here for couple of reasons. Firstly `row` is an object (and not a small one at that), so it's kind of annoying to make it work as a dependency in `useEffect` (I mean `JSONStringify`ing it _might_ be more expensive than just calling `prepareRow`). Secondly [react-table docs](https://react-table-v7.tanstack.com/docs/api/useTable) actually explicitly say to call this function before every render (I sure hope they optimized it in some way haha). And also we already call `prepareRow` each time in [default view](https://github.com/openedx/paragon/blob/2e7847f4b1bda995311edd50d86fff548ec7bdd8/src/DataTable/Table.jsx#L19) and if that hasn't caused any issues I doubt `CardView`

Reply:
Perfect! Thanks for digging into this.